go/vt/wrangler/reparent.go

```diff
@@ -411,6 +411,10 @@ func (wr *Wrangler) plannedReparentShardLocked(ctx context.Context, ev *events.R
 	defer remoteCancel()
 
 	var reparentJournalPos string
+	oldMasterTabletInfo, ok := tabletMap[topoproto.TabletAliasString(shardInfo.MasterAlias)]
```

Comment:
Was the previous location for this check causing a problem?

I had intentionally put this inside the `else{}` block because otherwise we're creating a variable called `oldMasterTabletInfo` that might actually refer to the new master, in the case when we are retrying PRS to the current master.

I was worried that could lead to bugs down the line, like what happened in TER when we similarly started allowing the code to continue even if `oldMaster == newMaster`.

Note that we already check that the new master is in the tablet map up above.